### PR TITLE
The Modules and Promises Section

### PR DESCRIPTION
**Pull request by Moyo75**
I did the modules and promises section of the project.

### CODE REVIEW

File: project-javascript-d1-functions/JohnAdemoye.html [hunk 1]
```diff
@@ -51,7 +51,7 @@
 
       // 6
       function formatDate(year, month, day) {
-        return year + '/' + convert2String(month) + '/' + convert2String(day);
+        return `${year}/${convert2String(month)}/${convert2String(day)}`;
```
Comment (basir) [inline project-javascript-d1-functions/JohnAdemoye.html]: Good

File: project-javascript-k1-advanced-objects/JohnAdemoye.html [hunk 2]
```diff
@@ -23,7 +23,7 @@
           if (!this.firstName) {
             return `Mr/Mrs ${this.lastName}`;
           }
-          return `Mr/Mrs ${this.firstName} ${this.lastName}`;
+          return `${this.firstName} ${this.lastName}`;
```
Comment (basir) [inline project-javascript-k1-advanced-objects/JohnAdemoye.html]: Nice job

File: project-javascript-l1-classes/JohnAdemoye.html [hunk 3]
```diff
@@ -33,8 +33,8 @@
 
       // 8/9
       class VIPTrafficLight extends TrafficLight {
-        constructor(SpecialEffects) {
-          super(SpecialEffects)
+        constructor(color, address, SpecialEffects) {
+          super(color, address)
```
Comment (basir) [inline project-javascript-l1-classes/JohnAdemoye.html]: Good

File: project-javascript-n1-modules/JohnAdemoye/src/index.js [hunk 4]
```diff
@@ -1,3 +1,15 @@
-console.log('Hello Node.js');
-import os from 'os';
-console.log('Platform: ' + os.platform());
+import { food, getSpecials } from './food';
```
Comment (basir) [inline project-javascript-n1-modules/JohnAdemoye/src/index.js]: Nice import

File: project-javascript-n1-modules/Oluwaseyi/src/food.js [hunk 5]
```diff
@@ -1,17 +1,17 @@
 const food = [
-  "Saffron",
-  "Kale",
-  "Clementine",
-  "Peaches",
-  "Nori",
-  "Kobe",
-  "Olive",
-  "Rosemary"
+  'Saffron',
```
Comment (basir) [inline project-javascript-n1-modules/Oluwaseyi/src/food.js]: Do not change other students code.

File: project-javascript-o1-promises/JohnAdemoye.html [hunk 6]
```diff
@@ -0,0 +1,47 @@
+<!DOCTYPE html>
+<html>
+  <body>
+    <script>
+      // your code here
+
+      // 3
+      const wash = (resolve, reject) => {
+        setTimeout(function() {
```
Comment (basir) [inline project-javascript-o1-promises/JohnAdemoye.html]: Nice one

File: project-javascript-o1-promises/JohnAdemoye.html [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+          } else {
+            reject('Wash: Fail');
+          }
+
+          // console.log('Value of numDishes is ' + numDishes);
+        }, 1000);
+      };
+
+      // 4
+      let myPromise = new Promise(wash);
```
Comment (basir) [inline project-javascript-o1-promises/JohnAdemoye.html]: could use const

File: project-javascript-n1-modules/Oluwaseyi/src/food.js (hunk 5 above)
Comment (Moyo75) [reply project-javascript-n1-modules/Oluwaseyi/src/food.js]: I never intended to. I think it was my prettier formatting.

File: project-javascript-o1-promises/JohnAdemoye.html (hunk 7 above)
Comment (Moyo75) [reply project-javascript-o1-promises/JohnAdemoye.html]: OK. Thanks.

File: project-javascript-o1-promises/JohnAdemoye.html (hunk 6 above)
Comment (Moyo75) [reply project-javascript-o1-promises/JohnAdemoye.html]: Thanks

File: project-javascript-n1-modules/JohnAdemoye/src/index.js (hunk 4 above)
Comment (Moyo75) [reply project-javascript-n1-modules/JohnAdemoye/src/index.js]: Thanks

File: project-javascript-l1-classes/JohnAdemoye.html (hunk 3 above)
Comment (Moyo75) [reply project-javascript-l1-classes/JohnAdemoye.html]: Thanks

File: project-javascript-k1-advanced-objects/JohnAdemoye.html (hunk 2 above)
Comment (Moyo75) [reply project-javascript-k1-advanced-objects/JohnAdemoye.html]: Thanks

File: project-javascript-d1-functions/JohnAdemoye.html (hunk 1 above)
Comment (Moyo75) [reply project-javascript-d1-functions/JohnAdemoye.html]: Thanks

**Comment by basir** [comment]
Hello John,

I do not have your email to let you know when the new project are ready.
Please give me your email.

regards,
Basir

**Comment by Moyo75** [comment]
Oh! My email address is moyojohn188@gmail.com
Thanks

Best regards

**Pull request merged**